api/scpca_portal/models/project.py

```diff
@@ -745,6 +463,166 @@ def get_additional_terms(self):
+        }
+
+    def get_multiplexed_sample_libraries_mapping(self, multiplexed_libraries_metadata: List[Dict]):
+        """
+        Returns a dictionary with keys as sample ids
+         and values as all of the samples associated libraries
+        """
+        multiplexed_sample_library_mapping = {}  # Sample ID to library IDs mapping.
+        for library_metadata in multiplexed_libraries_metadata:
+            multiplexed_library_sample_ids = library_metadata["demux_samples"]
+            for multiplexed_sample_id in multiplexed_library_sample_ids:
+                # Populate multiplexed library mapping.
+                if multiplexed_sample_id not in multiplexed_sample_library_mapping:
+                    multiplexed_sample_library_mapping[multiplexed_sample_id] = set()
+                multiplexed_sample_library_mapping[multiplexed_sample_id].add(
+                    library_metadata["scpca_library_id"]
+                )
+
+        return multiplexed_sample_library_mapping
+
+    def get_multiplexed_with_mapping(self, multiplexed_libraries_metadata: List[Dict]):
+        """
+        Return a dictionary with keys being specific demux ids,
+        and the values being all ids that this sample was multiplexed with.
+        """
+        multiplexed_with_mapping = {}
+        for library_metadata in multiplexed_libraries_metadata:
+            multiplexed_library_sample_ids = library_metadata["demux_samples"]
+            for multiplexed_sample_id in multiplexed_library_sample_ids:
+                # Remove sample ID from a mapping as sample cannot be
+                # multiplexed with itself.
+                multiplexed_library_sample_ids_copy = set(multiplexed_library_sample_ids)
+                multiplexed_library_sample_ids_copy.discard(multiplexed_sample_id)
+
+                # Populate multiplexed sample mapping.
+                if multiplexed_sample_id not in multiplexed_with_mapping:
+                    multiplexed_with_mapping[multiplexed_sample_id] = set()
+                multiplexed_with_mapping[multiplexed_sample_id].update(
+                    multiplexed_library_sample_ids_copy
+                )
+
+        return multiplexed_with_mapping
+
+    def get_multiplexed_with_combined_metadata(
+        self,
+        scpca_sample_id,
+        multiplexed_with_mapping,
+        multiplexed_sample_libraries_mapping,
+        samples_metadata_filtered_keys,
+        libraries_metadata_filtered_keys,
+    ):
+        """
+        Returns a list of combined metadata of all samples that
+        the scpca_sample_id sample was multiplexed with.
+        The returned list is then immediately written to the open csv file in the caller method.
+        """
+        multiplexed_with_combined_metadata = []
+        multiplexed_sample_ids = sorted(multiplexed_with_mapping[scpca_sample_id])
+
+        multiplexed_with_samples_metadata_filtered_keys = {
```

Comment:
This name is pretty hard to parse without looking at how it is used. I think an inline comment describing the variable's value would be prudent here.